bootstrap/051_materialization_status.sql

```diff
@@ -0,0 +1,100 @@
+
+create or replace function internal.success_to_status(success boolean) returns text
+as
+$$
+    IFF(success IS NULL, NULL, IFF(success, 'SUCCESS', 'FAILURE'))
+$$;
+
+create or replace function internal.task_complete(state text) returns boolean
+as
+$$
+    state in ('SUCCEEDED', 'FAILED', 'FAILED_AND_AUTO_SUSPENDED')
+$$;
+
+create or replace function internal.task_pending(state text) returns boolean
+as
+$$
+    state in ('SCHEDULED', 'EXECUTING')
+$$;
+
+
+create or replace view admin.materialization_status
+as
+-- The tables we materialize and how often the task refreshes them.
+with task_tables as (
+    select table_name, user_schema, user_view, task_name from ( values
+        ('QUERY_HISTORY', 'REPORTING', 'ENRICHED_QUERY_HISTORY', 'QUERY_HISTORY_MAINTENANCE'),
+        ('WAREHOUSE_EVENTS_HISTORY', 'REPORTING', 'WAREHOUSE_SESSIONS', 'WAREHOUSE_EVENTS_MAINTENANCE'),
+        ('WAREHOUSE_EVENTS_HISTORY', 'REPORTING', 'CLUSTER_SESSIONS', 'WAREHOUSE_EVENTS_MAINTENANCE')
+    ) as t(table_name, user_schema, user_view, task_name)
+), task_history as (
+    -- The history rows generated every time a task is run and the table that task is managing
+    select $1 as run, $2 as success, $3 as input, $4 as output, $5 as table_name, $6 as task_name, IFF(INPUT is null, 'FULL', 'INCREMENTAL') as kind FROM (
+        select run::TIMESTAMP_LTZ, success, input, output, 'QUERY_HISTORY', 'QUERY_HISTORY_MAINTENANCE'   from internal.task_query_history
+        union all
+        select run::TIMESTAMP_LTZ, success, input, output, 'WAREHOUSE_EVENTS_HISTORY', 'WAREHOUSE_EVENTS_MAINTENANCE' from internal.task_warehouse_events
+        union all
+        select run::TIMESTAMP_LTZ, success, input, output, table_name, 'SIMPLE_DATA_EVENTS_MAINTENANCE' from internal.task_simple_data_events
+    )
+), sf_task_history as (
+    select name as task_name, query_id, graph_run_group_id, state, scheduled_time, QUERY_START_TIME,
+    from table(information_schema.task_history(SCHEDULED_TIME_RANGE_START => TIMESTAMPADD(HOUR, -3, current_timestamp()), result_limit => 1000))
+    WHERE database_name in (select current_database()) and schema_name = 'TASKS'
+    union
```

Comment:
Creating the view inside the setup script fails:

```
File: 051_materialization_status.sql
[Uncaught exception of type 'STATEMENT_ERROR' on line 21 at position 0 : Stored procedure execution error: Requested information on the current user is not accessible in stored procedure.] was primary error.
```

I'm guessing because of the setup script limitations rather than a native app limitation. https://docs.snowflake.com/en/developer-guide/native-apps/adding-application-logic#limitations-when-calling-context-functions-from-the-setup-script

Reply:
Jacques to the rescue -- have to create it from within a task in the native app. Same sort of thing that probe_monitoring is doing.